### PR TITLE
Feature/17 password change

### PR DESCRIPTION
**Pull request by AntonioRodriguezRuiz**
This pull request introduces a new feature allowing users to change their passwords. The changes include updates to the user controller, route definitions, and validation logic.

### New Feature: Password Change

* Added `changePassword` function to handle password change logic in `src/controllers/userController.js`. This includes verifying the current password, updating to a new password, and logging the operation.
* Updated `src/routes/userRoutes.js` to include the new `changePassword` route and its associated middleware for validation and authorization. [[1]](diffhunk://#diff-5c0d80e83b24f65ac31f45489b83a31f574e99a6c3dcea0a6ae91ed2f06e8161R6) [[2]](diffhunk://#diff-5c0d80e83b24f65ac31f45489b83a31f574e99a6c3dcea0a6ae91ed2f06e8161R16) [[3]](diffhunk://#diff-5c0d80e83b24f65ac31f45489b83a31f574e99a6c3dcea0a6ae91ed2f06e8161R46-R54)
* Added `changePasswordValidator` in `src/validators/userValidator.js` to ensure the new password meets security requirements.

### CODE REVIEW

File: src/routes/userRoutes.js [hunk 1]
```diff
@@ -41,6 +43,14 @@ router.put(
   editUser
 );
 
+router.post(
+  '/users/change-password',
+  validateToken,
+  validate(changePasswordValidator),
+  authorizeRequest('changePassword'),
+  changePassword
+);
```
Comment (alvgonfri) [inline src/routes/userRoutes.js:52]: Wouldn't this be a PUT method?

File: src/controllers/userController.js [hunk 2]
```diff
@@ -140,6 +140,44 @@ export const editUser = async (req, res) => {
...
+        url: req.originalUrl,
+        userId: userId,
+      });
+      return res.status(400).json({ message: 'Invalid credentials' });
+    }
+
+    user.password = newPassword;
+    await user.save();
+
+    deleteTokensByUserId(userId, req.cookies.token);
```
Comment (alvgonfri) [inline src/controllers/userController.js:162]: If the token is deleted, you are not supposed to be able to perform operations that require being logged in, right?

I have seen that I can execute several consecutive password changes without having to log in again.

**Review by alvgonfri — CHANGES_REQUESTED** [review]

File: src/controllers/userController.js (hunk 2 above)
Comment (AntonioRodriguezRuiz) [reply src/controllers/userController.js:162]: Yes that is correct. The token does not expire for the person who calls the method. You can try opening two sessions and you will see the other session is expired. The second parameter of the method is to exclude the token

File: src/routes/userRoutes.js (hunk 1 above)
Comment (AntonioRodriguezRuiz) [reply src/routes/userRoutes.js:52]: According to online documentation, it should be a post

**Review by alvgonfri — APPROVED** [review]
The feature work correclty.

**Pull request merged**